tests/sentry/integrations/opsgenie/test_integration.py

```diff
@@ -0,0 +1,94 @@
+from urllib.parse import urlencode
+
+import pytest
+import responses
+
+from sentry.integrations.opsgenie.integration import OpsgenieIntegrationProvider
+from sentry.models.integrations.integration import Integration
+from sentry.shared_integrations.exceptions import IntegrationError
+from sentry.testutils import IntegrationTestCase
+
+
+class OpsgenieIntegrationTest(IntegrationTestCase):
+    provider = OpsgenieIntegrationProvider
+    config = {"base_url": "https://api.opsgenie.com/", "api_key": "123"}
+
+    def setUp(self):
+        super().setUp()
+        self.init_path_without_guide = f"{self.init_path}?completed_installation_guide"
+
+    def assert_setup_flow(self, name="cool-name"):
+        resp = self.client.get(self.init_path)
+        assert resp.status_code == 200
+
+        resp = self.client.get(self.init_path_without_guide)
+        assert resp.status_code == 200
+
+        resp = self.client.post(self.init_path_without_guide, data=self.config)
+        assert resp.status_code == 200
+        responses.add(
+            responses.GET,
+            url="{}{}".format(self.config["base_url"].rstrip("/"), "/v2/account"),
+            json={"data": {"name": name}},
+        )
+
+        resp = self.client.get("{}?{}".format(self.setup_path, urlencode({"name": name})))
+
+        mock_api_key_verification = responses.calls[0].request
+        assert mock_api_key_verification.headers["Authorization"] == "GenieKey " + self.config.get(
+            "api_key"
+        )
```

Comment:
mypy is complaining here b/c `...get("api_key")` is of type `Optional[str]` so you could be adding a str and None which would error. You can assert the key you're getting before `+` to make sure the string is set
```suggestion
        assert mock_api_key_verification.headers["Authorization"] == "GenieKey " + self.config.get(
            "api_key"
        )
```